ThunderBasics/MultipleShadowView.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  MultipleShadowView.swift
+//  ShadowView
+//
+//  Created by Simon Mitchell on 23/07/2020.
+//  Copyright © 2020 3 Sided Cube. All rights reserved.
+//
+
+import UIKit
+
+
+protocol CornerObservableLayerDelegate: class {
+    
+    func cornerObservableLayer(_ layer: CornerObservableLayer, didUpdateCornerRadius cornerRadius: CGFloat)
+    
+    func cornerObservableLayer(_ layer: CornerObservableLayer, didUpdateCornerCurve cornerCurve: CALayerCornerCurve)
+}
+
+internal class CornerObservableLayer: CALayer {
+    
+    /// Delegate for corner property changes
+    weak var cornerDelegate: CornerObservableLayerDelegate?
+    
+    override var cornerRadius: CGFloat {
+        didSet {
+            cornerDelegate?.cornerObservableLayer(self, didUpdateCornerRadius: cornerRadius)
+        }
+    }
+    
+    override var cornerCurve: CALayerCornerCurve {
+        didSet {
+            cornerDelegate?.cornerObservableLayer(self, didUpdateCornerCurve: cornerCurve)
+        }
+    }
+}
+
+/// A subclass of `UIView` which ensures multiple shadows have their
+/// bounds, and corner radius properties updated when the parent view's properties
+/// change.
+public class MultipleShadowView: UIView {
+    
+    public override class var layerClass: AnyClass {
+        return CornerObservableLayer.self
+    }
+    
+    /// Whether the shadow layers should update their corner radius to match the view's
+    /// layer's corner radius
+    public var matchShadowCornerRadius: Bool = true {
+        didSet {
+            guard matchShadowCornerRadius else { return }
+            layer.sublayers?.filter({ $0 is ShadowLayer }).forEach({ (shadowLayer) in
+                shadowLayer.cornerRadius = layer.cornerRadius
+            })
+        }
+    }
+    
+    /// Whether the shadow layers should update their corner curve to match the view's
+    /// layer's corner curve
+    public var matchShadowCornerCurve: Bool = true {
+        didSet {
+            guard matchShadowCornerCurve else { return }
+            layer.sublayers?.filter({ $0 is ShadowLayer }).forEach({ (shadowLayer) in
+                if #available(iOS 13.0, *) {
+                    shadowLayer.cornerCurve = layer.cornerCurve
+                }
+            })
+        }
+    }
+    
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        (layer as? CornerObservableLayer)?.cornerDelegate = self
+    }
+    
+    public required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        (layer as? CornerObservableLayer)?.cornerDelegate = self
+    }
+
+    override public func layoutSubviews() {
+        super.layoutSubviews()
+        layer.sublayers?.filter({ $0 is ShadowLayer }).forEach({ (shadowLayer) in
+            shadowLayer.frame = bounds
```

Comment:
Do we want shadowPath? `frame` very well might be preferred, be interested to see what you think

Reply:
I think `frame` here 😄 `shadowPath` will give us `continuous` corners I believe (If you're thinking of using `UIBezierPath.roundedRect` but the way it is now, the user has complete control over the corner radii :)

Reply:
Sure let's do `frame` then :)